modules/tinymce/src/core/main/ts/dom/Empty.ts

```diff
@@ -1,85 +1,122 @@
-import { Fun } from '@ephox/katamari';
-import { Compare, ContentEditable, SelectorExists, SugarElement, Traverse } from '@ephox/sugar';
+import { Fun, Obj, Type } from '@ephox/katamari';
+import { Compare, ContentEditable, SugarNode, SugarElement, Traverse, PredicateExists } from '@ephox/sugar';
 
 import DomTreeWalker from '../api/dom/TreeWalker';
-import * as CaretCandidate from '../caret/CaretCandidate';
-import { isWhitespaceText } from '../text/Whitespace';
+import Schema from '../api/html/Schema';
+import { isWhitespaceText, isZwsp } from '../text/Whitespace';
 import * as NodeType from './NodeType';
 
-const hasWhitespacePreserveParent = (node: Node, rootNode: Node): boolean => {
-  const rootElement = SugarElement.fromDom(rootNode);
-  const startNode = SugarElement.fromDom(node);
-  return SelectorExists.ancestor(startNode, 'pre,code', Fun.curry(Compare.eq, rootElement));
+interface IsEmptyOptions {
```

Comment:
So this gets exposed to the public DOMUtils.isEmpty API. Is that something we want here or do we just want to expose a subset of these things? Before only `includeZwsp` was exposed as a option.
If we expose them then maybe we need to describe that we added them in the changelog or possibly even document them.